react/ProductDescription.tsx

```diff
@@ -55,14 +66,47 @@ function ProductDescription(props: PropsWithChildren<Props>) {
   const { handles } = useCssHandles(CSS_HANDLES, { classes: props.classes })
   const intl = useIntl()
   const { product } = useProduct()
+  const { isMobile } = useDevice()
 
+  const { collapseContent = true, title, customCollapse = false } = props
   const description = props.description ?? product?.description
+  const [isCustomCollapsed, setCustomCollapsed] = useState(customCollapse)
+
+  const {
+    desktopLimitCharacters = defaultDesktopCharacters,
+    mobileLimitCharacters = defaultMobileCharacters,
+  }: any = props
+
+  const desktopLimitChars: any = useRef(desktopLimitCharacters)
+  const mobileLimitChars: any = useRef(mobileLimitCharacters)
```

Comment:
I don't think you need that. You can just declare `const` variables inside the hook.

Reply:
![err](https://user-images.githubusercontent.com/43498488/109263337-c03b0980-780b-11eb-9b01-801ecaa4b526.jpg)

I use that because of this warning. Let me know if I let the code as it is or I need to remove those lines.

Reply:
The problem your editor is pointing to is: you're trying to edit the outside variable inside the hook. Indeed that's not a good practice, but you don't actually need these updated values outside the hook, just the value you return (`limitCharacters`)

I'll leave a comment below so you can see a possible solution to this.